src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -1123,20 +1135,25 @@ namespace smt::noodler {
             tout << "Creating formula for conversions" << std::endl;
         );
 
-        // the resulting formula 
+        // the resulting formula
         LenNode result(LenFormulaType::AND);
         LenNodePrecision res_precision = LenNodePrecision::PRECISE;
 
         // collect all variables that substitute some string_var of some conversion
         std::tie(code_subst_vars, int_subst_vars) = get_vars_substituted_in_conversions();
 
         // create formula for each variable substituting some string_var in some code conversion
-        result.succ.push_back(get_formula_for_code_subst_vars(code_subst_vars));
+        LenNode code_subst_formula = get_formula_for_code_subst_vars(code_subst_vars);
+        if (!code_subst_formula.succ.empty()) {
+            result.succ.push_back(code_subst_formula);
+        }
 
         // create formula for each variable substituting some string_var in some int conversion
         std::map<BasicTerm,std::vector<unsigned>> int_subst_vars_to_possible_valid_lengths;
         auto int_conv_formula_with_precision = get_formula_for_int_subst_vars(int_subst_vars, code_subst_vars, int_subst_vars_to_possible_valid_lengths);
-        result.succ.push_back(int_conv_formula_with_precision.first);
+        if (!int_conv_formula_with_precision.first.succ.empty()) {
+            result.succ.push_back(int_conv_formula_with_precision.first);
+        }
```

Comment:
Why were these checks added? To make it more readable? Because I think it should work even without them.

Reply:
Yes, it will work. But when you want to dump `LenNode` into SMT2, you cannot have empty children, so it is mostly for "sanitization of the internal AST".

Reply:
Ah ok, that makes sense.